initializers/database.go

```diff
@@ -60,24 +64,77 @@ func connectSqlite() (*gorm.DB, error) {
 }
 
 func connectPostgres() (*gorm.DB, error) {
-	dsn := fmt.Sprintf(
-		"host=%s user=%s password=%s dbname=%s port=%d %s",
-		config.DB_POSTGRES_HOST,
-		config.DB_POSTGRES_USERNAME,
-		config.DB_POSTGRES_PASSWORD,
-		config.DB_POSTGRES_DB_NAME,
-		config.DB_POSTGRES_PORT,
-		config.DB_POSTGRES_DSN_EXTRA,
-	)
+	var dsn string
+	if config.DB_POSTGRES_DSN != "" {
+		dsn = config.DB_POSTGRES_DSN
+	} else {
+		dsn = fmt.Sprintf(
+			"host=%s user=%s password=%s dbname=%s port=%d %s",
+			config.DB_POSTGRES_HOST,
+			config.DB_POSTGRES_USERNAME,
+			config.DB_POSTGRES_PASSWORD,
+			config.DB_POSTGRES_DB_NAME,
+			config.DB_POSTGRES_PORT,
+			config.DB_POSTGRES_DSN_EXTRA,
+		)
+	}
 
 	return gorm.Open(
 		postgres.Open(dsn),
 		&gorm.Config{},
 	)
 }
 
+func connectMysql() (*gorm.DB, error) {
+	var dsn string
+	if config.DB_MYSQL_DSN != "" {
+		dsn = config.DB_MYSQL_DSN
+	} else {
+		dsn = fmt.Sprintf(
+			"%s:%s@tcp(%s:%d)/%s",
+			config.DB_MYSQL_USERNAME,
+			config.DB_MYSQL_PASSWORD,
+			config.DB_MYSQL_HOST,
+			config.DB_MYSQL_PORT,
+			config.DB_MYSQL_DB_NAME,
+		)
+		if config.DB_MYSQL_DSN_EXTRA != "" {
+			dsn += fmt.Sprintf(
+				"?%s",
+				config.DB_MYSQL_DSN_EXTRA,
+			)
+		}
+	}
+
+	return gorm.Open(
+		mysql.Open(dsn),
+		&gorm.Config{},
+	)
+}
+
+func migrateTable(model any, table_name string) {
+	err := DB.AutoMigrate(model)
+	if err != nil {
+		log.Error().Err(err).Str(
+			"table_name",
+			table_name,
+		).Msg(
+			"Could not AutoMigrate table",
+		)
+	}
+}
+
 func migrateTables() {
```

Comment:
I Know it's not new but now we need to think about a single source of truth for the db schema, if it's go then we need to also use the CRUD in the UI